tasks.py

```diff
@@ -96,20 +96,18 @@ def update(c, check=False):
 @task
 def setup(c):
     """Set up the development environment."""
-    if which("pdm") or ACTIVE_VENV:
+    if UV:
+        c.run(f"{UV} self update", pty=PTY)
```

Comment:
Because there are [multiple ways to install `uv`](https://docs.astral.sh/uv/getting-started/installation/), I think it would be better to omit the `self update` step here, [which appears to be designed only for use with the standalone installer](https://docs.astral.sh/uv/getting-started/installation/#upgrading-uv). I cannot think of a generalized way to upgrade `uv` for folks who might have installed `uv` via Pipx, Cargo, Homebrew, etc.